File: Leetcode 2201-2300/2274/2274.py
# Leetcode 2274: Maximum Consecutive Floors Without Special Floors
# https://leetcode.com/problems/maximum-consecutive-floors-without-special-floors/
# Solved on 12th of November, 2025
class Solution:
    def maxConsecutive(self, bottom: int, top: int, special: list[int]) -> int:
        """
        Calculates the maximum number of consecutive floors without any special floors.

        Args:
            bottom (int): The lowest floor number.
            top (int): The highest floor number.
            special (list[int]): A list of special floor numbers.
        Returns:
            int: The maximum number of consecutive floors without special floors.
        """
        special.sort()

        maxFound = 0

        maxFound = max(maxFound, (special[0] - bottom))

        for i in range(1, len(special)):
            currentGap = special[i] - special[i - 1] - 1
            maxFound = max(maxFound, currentGap)

        maxFound = max(maxFound, (top - special[-1]))

        return maxFound

File: Leetcode 2201-2300/2274/2274.java
// Leetcode 2274: Maximum Consecutive Floors Without Special Floors
// https://leetcode.com/problems/maximum-consecutive-floors-without-special-floors/
// Solved on 12th of November, 2025
import java.util.Arrays;

class Solution {
    /**
     * Calculates the maximum number of consecutive floors without any special floors.
     *
     * @param bottom The lowest floor number.
     * @param top The highest floor number.
     * @param special An array of floor numbers that are considered special.
     * @return The maximum number of consecutive floors that are not special.
     */
    public int maxConsecutive(int bottom, int top, int[] special) {
        if (special == null || special.length == 0) {
            return top - bottom + 1;
        }
        Arrays.sort(special);
        int maxGap = 0;
        int firstGap = special[0] - bottom;
        if (firstGap > maxGap) {
            maxGap = firstGap;
        }
        for (int i = 1; i < special.length; i++) {
            int gap = special[i] - special[i - 1] - 1;
            if (gap > maxGap) {
                maxGap = gap;
            }
        }
        int lastGap = top - special[special.length - 1];
        if (lastGap > maxGap) {
            maxGap = lastGap;
        }
        return maxGap;
    }
}